### ISSUE TITLE
Implement jQuery for RPC calls

### ISSUE THREAD

**Issue by Nashluffy** [Nashluffy/SkyAudio#23]
Basically, the way we currently have it we would have to redirect or reload the page to update the music clip. We want to utilize AJAX through jQuery to update our DOM. 

See https://pythonprogramming.net/jquery-flask-tutorial/

**Comment by Nashluffy** [comment]
@pressure101 @leedagr8 this is a front end feature that I think is key right now. Like I said at our meeting, we can process the clip but have no way of displaying the processed clip.

**Comment by Nashluffy** [comment]
@leedagr8 Think this is a good link to look at. What we're doing classifies as a cross-origin request (CORS). I'm updating the Flask side of things to enable this.

https://www.codecademy.com/articles/what-is-cors

**Comment by Nashluffy** [comment]
I think what we'll end up doing is POST'ing the blob URL, download it via the URL on the server, process and then return.

**Comment by leedagr8** [comment]
@Nashluffy Found a cool website we could look at it. [Pizzicato API](https://alemangui.github.io/pizzicato/#dub-delay). May be worth looking at we progress.

**Comment by leedagr8** [comment]
@Nashluffy Hey man I found the return type is "basic". I'm trying to verify that the blob of data is being sent over to the server side. I believe it is. 

I really need us to sit down and work on this sunday morning before I go to work and try to figure this out. I'm not sure about a few things.

1.  Like are we sure that same file on the front end is going to the backend server. 
2.  Why is the backend returning a file that's "basic" and not of file ".wav"



![image](https://user-images.githubusercontent.com/26153230/65299506-cff5ef00-db3d-11e9-8948-e005fe594bfc.png)

**Comment by Nashluffy** [comment]
@leedagr8 I've verified the blob of data is being sent to the backend. I can show you if you want, but I thought we were working on the return of the file. The return of the file shouldn't have anything to do with whether or not the file is being sent to the server, other than it's wrapped in the same POST.

1. See above, but I can show you if you want

2. The response is of type basic, and is one of the 5 response types you can have. See [Response.type](https://developer.mozilla.org/en-US/docs/Web/API/Response/type) The fact that it's a basic response type is good! We just have to figure out how to parse the file from the response. It might be a .wav or it might be a .blob, and I'm not sure if we can convert a .wav back to a blob. So we might have to add handling for that in javascript, because our audio engine doesn't take a blob.

**Comment by leedagr8** [comment]
@Nashluffy I was looking at the type of data that was being sent to the backend so I wanted to verify that is was a blob of data. The reason I'm confused is because the response should be returning a blob or type "./wav" not basic. So I wasn't sure if we were actually sending POSTing the right blob of data to the backend, since the blob of data that's sent to the backend is the data that will be process.

**Comment by leedagr8** [comment]
@Nashluffy 
That link above really helped I'm going to test some things out. hopefully I can have something for you before I go to work tonight

**Comment by leedagr8** [comment]
@Nashluffy Hey dude like I mentioned in our text I'm able to retrieve the song from the backend :)! 

If you look in the Skyaudio.py file I changed the return from " return send_file(os.path.join(application.config['UPLOAD_FOLDER'], 'processed.wav'))"
To 
"send_file('/home/leedagr8/Downloads/02 Bored To Death.mp3', attachment_filename='Testing.ogg')"

Reason being, I needed to check and make sure the backend was successfully returning a song back from my local drive. so I downloaded one of our songs from our SD1 google drive folder [here](https://drive.google.com/drive/u/1/folders/1fbxjZa3oddwbLj6gTMldeOE8f78rnk4y )
     
But it worked successfully. Only thing I would need to do now is figure out how to return the name  f the audio file(that we're returning) to the front end.

But for now the hard part is done! Good work bro :+1: 

![image](https://user-images.githubusercontent.com/26153230/65334918-656e9e80-db91-11e9-8f05-aa0786a498b7.png)
